pkg/api/ui_handler.go

```diff
@@ -33,7 +34,8 @@ func NewUIHandler(gatewayDomains []string, snippets []params.CodeSnippet) http.H
 		panic(err)
 	}
 	fileSystem := http.FS(injectedContent)
-	etagHandler := EtagMiddleware(injectedContent, http.StripPrefix("/", http.FileServer(fileSystem)))
+	gzipHandler := gziphandler.GzipHandler(http.FileServer(fileSystem))
```

Comment:
Not sure that I understand.  We are serving just a few bundles, they are statically embedded into the executable.  Can we hold the precompress the bundled objects and serve them directly but compressed?  (Does [any browser from the past 10 years](https://caniuse.com/?search=gzip) _not_ support this?)

Reply:
You're not wrong. However, it's unusual to store GZipped assets on disk. GZipping the response is usually the responsibility of some reverse proxy in front of the asset server (e.g., nginx, Varnish, or something like CloudFront.)  
I'll concede that most reasons for doing this aren't relevant to our use case. However, I decided to go this route under these assumptions:

1. Low complexity
2. Static assets can be cached client side and we can expect a low request rate in regular use
3. The Go impl. isn't as efficient as the one used by nginx (C-based), but not far from it and we're using the default level, so this isn't expected to create any noticeable additional compute load